lib/rules/td-headers-attr.json

```diff
@@ -16,8 +16,8 @@
   ],
   "actIds": ["a25f45"],
   "metadata": {
-    "description": "Ensure that each cell in a table that uses the headers attribute refers only to other cells in that table",
-    "help": "Table cells that use the headers attribute must only refer to cells in the same table"
+    "description": "Ensure that each cell in a table using the headers attribute refers only to header cells within that table",
+    "help": "Table cells using the headers attribute must refer only to header cells or data cells with the role=columnheader/rowheader within the same table."
```

Comment:
I appreciate the attempt to make this more concrete for folks that might not know exactly what a "header cell" formally means. But for consistency with other rules, it's actually preferable for "help" to be the shorter one rather than "description" (the fields are poorly named; "help" is really more like "suggested tooltip", so we try to keep it more concise than this). For both of them, we try to keep the overall lengths consistent between rules and fairly concise; the old help text was already towards the wordier end of what we prefer.

But I do agree that it's more understandable to use precise element/attribute names. I think we can actually get away with not mentioning the "role=" option in the text here; we only allow headers to point to a non-`<th>` element with header roles because it happens to be accessibility supported (ie, not a WCAG violation), but it's not actually compliant with the HTML standard ([which requires pointing specifically to `<th>` elements](https://html.spec.whatwg.org/multipage/tables.html#attributes-common-to-td-and-th-elements)), so I don't think we need the help text to encourage it as an option. In general, rule descriptions/help can and should omit mention of edge cases that we don't encourage but technically allow.

```suggestion
    "description": "Ensure that each cell in a table that uses the headers attribute refers only to other <th> elements in that table",
    "help": "Table cell headers attributes must refer to other <th> elements in the same table."
```

Reply:
Got your point. When I updated this text used following rule as example https://github.com/engineerklimov/axe-core/blob/develop/lib/rules/th-has-data-cells.json. You correctly undestood my intention behind message text, wanted to be 100% explicit about pass conditions for that check.